packages/core-utils/src/httpie.ts

```diff
@@ -94,7 +94,7 @@ class Httpie {
             const { body, headers, statusCode } = await got[method](url, opts);
 
             return {
-                body: JSON.parse(body || {}),
+                body: JSON.parse(body),
```

Comment:
```suggestion
                body: JSON.parse(body || "{}"),
```

In its current state it will throw if `body` is empty or `undefined` since parsing will fail, which is always the case when the response is to a `HEAD` request. In real terms this means port pinging in the peer communicator will set all port values to `-1` in the `/api/peers` endpoint.